main.go

```
@@ -340,6 +377,26 @@ func localPathExists(path string) bool {
 	return err == nil
 }
 
+// resolveSource 解析源包目录：优先 ailuo.toml；found=false 表示无 ailuo.toml
+// （由旧 manifest.json 路径处理）。若清单声明 [build]，先执行构建再返回
+// （pack/publish 共用，构建失败即报错，不打包残缺工件）。
+func resolveSource(ctx context.Context, sourceDir string) (manifest packmgr.Manifest, manifestBytes []byte, found bool, err error) {
+	path := packagefmt.SourcePath(sourceDir)
+	if _, statErr := os.Stat(path); statErr != nil {
+		return packmgr.Manifest{}, nil, false, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 385) but it returns nil

(nilerr)